### DISCUSSION TITLE
Renovate-config-validator cannot validate config.js if hostRules token is read from environment variable

### DISCUSSION THREAD

**Discussion by wcarlsen** [renovatebot/renovate#29204] [Request Help]
### What would you like help with?

I think I found a bug

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Github

### Please tell us more about your question or problem

Renovate-config-validator cannot validate config.js if hostRules token is read from environment variable.

My config.js looks something like this:

```javascript
module.exports = {
  ...
  hostRules: [
    {
      hostType: 'terraform-module',
      matchHost: 'https://app.terraform.io',
      token: process.env.RENOVATE_TF_CLOUD_TOKEN,
    },
  ],
  ...
}
```
When I run `renovate-config-validator` i get the following error:

```bash
INFO: Validating config.js
 WARN: Found errors in configuration
       "file": "config.js",
       "warnings": [
         {
           "topic": "Configuration Error",
           "message": "Configuration option `hostRules[0].token` should be a string."
         }
       ]
```

This means that for use cases such as pre-commit-hooks it is broken.

Lastly big thank you for a great project, that get so much joy from!

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment]
I don't think this is true. I just replicated your hostRules and substituted for a valid env variable and it validated successfully. When I put in an env variable which didn't exist, then I got the same error as you. I think this is related to your environment somehow

**Reply by wcarlsen** [reply]
Hmm yeah I get that. I think I then described this poorly and that's my bad. When validating the config.js the token is not yet known. But maybe that is required then?

**Reply by rarkins** [reply]
Yes. So put a dummy value into env if you need

**Reply by wcarlsen** [reply]
Thanks!
